components/apimgt/org.wso2.carbon.apimgt.impl/src/test/java/org/wso2/carbon/apimgt/impl/definitions/AsyncApiParserTest.java

```diff
@@ -0,0 +1,65 @@
+/*
+ *  Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.wso2.carbon.apimgt.impl.definitions;
+
+import org.apache.commons.io.IOUtils;
+import org.junit.Assert;
+import org.junit.Test;
+import org.wso2.carbon.apimgt.api.model.API;
+import org.wso2.carbon.apimgt.api.model.APIIdentifier;
+
+import java.io.File;
+import java.util.HashMap;
+import java.util.Map;
+
+public class AsyncApiParserTest {
+
+    @Test
+    public void testGenerateAsyncAPIDefinition() throws Exception {
```

Comment:
@Chamindu36 Did we write test for failure scenarios as well ? Whether we throw the exceptions etc

Reply:
Fixed with https://github.com/wso2/carbon-apimgt/pull/10842/commits/c41ba333ee522fd29e341c130570450703f787bf